src/compiler.jl

```diff
@@ -246,12 +243,16 @@ function emit_allocobj!(B, T, size)
 	T_jlvalue = LLVM.StructType(LLVMType[]; ctx)
     T_prjlvalue = LLVM.PointerType(T_jlvalue, #= AddressSpace::Tracked =# 10)
 
-	ty = inttoptr!(B, LLVM.ConstantInt(convert(Int, pointer_from_objref(T)); ctx), LLVM.PointerType(T_jlvalue))
-	ty = addrspacecast!(B, ty, T_prjlvalue)
+    T_ppjlvalue = LLVM.PointerType(LLVM.PointerType(T_jlvalue))
+
+	ty = LLVM.const_inttoptr(LLVM.ConstantInt(convert(Int, pointer_from_objref(T)); ctx), LLVM.PointerType(T_jlvalue))
+	ty = LLVM.const_addrspacecast(ty, T_prjlvalue)
+
+    pgcstack = reinsert_gcmarker!(fn, B)
+    ct = inbounds_gep!(B, bitcast!(B, pgcstack, T_ppjlvalue), [LLVM.ConstantInt(current_task_offset(); ctx)])
+
 	size = LLVM.ConstantInt(T_size, size)
-    args = [reinsert_gcmarker!(fn), size, ty]
-	args[1] = bitcast!(B, args[1], parameters(eltype(llvmtype(func)))[1])
-    return call!(B, func, args)
+    return call!(B, func, [ct, size, ty])
```

Comment:
You probably need to branch on versions here since some take the ct and some the ptls. See vc/rules2 for a full implementation. Also see the corrected constants there for `current_task_offset`

Reply:
This is taken from vc/rules2 (or was attempted to).

Reply:
Sorry all the PRs blend together.

What you want is https://github.com/EnzymeAD/Enzyme.jl/blob/410a7cc75a0d5fa04e6f2e75a931020e0434f1f9/src/compiler.jl#L3702-L3710
and https://github.com/EnzymeAD/Enzyme.jl/blob/410a7cc75a0d5fa04e6f2e75a931020e0434f1f9/src/compiler.jl#L3732-L3736 + https://github.com/EnzymeAD/Enzyme.jl/blob/410a7cc75a0d5fa04e6f2e75a931020e0434f1f9/src/compiler.jl#L3718-L3729

and if you really want to make me happy you merge that into `emit_allocobj` so that there is one correct implementation.